frontend/class-frontend.php

```diff
@@ -1226,8 +1228,16 @@ public function metakeywords() {
 		 */
 		$keywords = apply_filters( 'wpseo_metakeywords', trim( $keywords ) ); // More appropriately named.
 
-		if ( is_string( $keywords ) && $keywords !== '' ) {
-			echo '<meta name="keywords" content="', esc_attr( strip_tags( stripslashes( $keywords ) ) ), '"/>', "\n";
+		if ( $echo !== false ) {
```

Comment:
I'd like to propose the following alterations:

```
private function sanitize_keywords( $keywords ) {
     if ( is_string( $keywords ) && $keywords !== '' ) {
          return esc_attr( strip_tags( stripslashes( $keywords ) ) );
    }

    return '';
}
```

Followed by altering the code as such:
```
$sanitized_keywords = $this->sanitize_keywords( $keywords );

if ( $echo === false  ) {
      return $sanitized_keywords;
}

echo '<meta name="keywords" content="', $sanitized_keywords, '"/>', "\n";
```

This should cover all situations and improve the readability greatly :)

If anything is unclear about why I suggested this, just let me know and I'll gladly elaborate further :)